diff -urN mlocate/doc/locate.1.in mlocate-0.22.2/doc/locate.1.in
--- mlocate/doc/locate.1.in	2008-03-02 06:02:20.000000000 +0100
+++ mlocate-0.22.2/doc/locate.1.in	2012-09-24 14:03:57.617354099 +0200
@@ -189,7 +189,7 @@
 
 .TP
 \fB\-V\fR, \fB\-\-version\fR
-Write information about the version and licence of
+Write information about the version and license of
 .B locate
 on standard output and exit successfully.
 
