diff -urN mlocate/src/updatedb.c mlocate-0.22.2/src/updatedb.c
--- mlocate/src/updatedb.c	2012-09-24 14:00:00.925995671 +0200
+++ mlocate-0.22.2/src/updatedb.c	2012-09-24 14:09:57.455418405 +0200
@@ -773,6 +773,15 @@
       if (strcmp (de->d_name, ".") == 0 || strcmp (de->d_name, "..") == 0)
 	continue;
       name_size = strlen (de->d_name) + 1;
+      if (name_size == 1)
+	{
+	  /* Unfortunately, this does happen, and mere assert() does not give
+	     users enough information to complain to the right people. */
+	  error (0, 0,
+		 _("file system error: zero-length file name in directory %s"),
+		 dest->path);
+	  continue;
+	}
       assert (name_size > 1);
       entry_size = offsetof (struct entry, name) + name_size;
       if (entry_size > OBSTACK_SIZE_MAX)
